recipes/imake/all/conanfile.py

```diff
@@ -109,32 +100,40 @@ def _configure_autotools(self):
             "--enable-revpath={}".format(yes_no(self.options.revpath)),
             "--enable-xmkmf={}".format(yes_no(self.options.xmkmf)),
         ]
-
-        # FIXME: RAWCPP (ac_cv_path_RAWCPP) is not compatible with MSVC preprocessor. It needs to be cpp.
-        if tools.get_env("CPP"):
+        if "CPP" in os.environ:
             conf_args.extend([
-                "--with-script-preproc-cmd={}".format(tools.get_env("CPP")),
+                "--with-script-preproc-cmd={}".format(os.environ["CPP"]),
             ])
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+
+        env = tc.environment()
+        if is_msvc(self):
+            compile_wrapper = f"{unix_path(self, self.conf.get('user.automake:compile-wrapper'))} cl -nologo"
+            env.define("CC", compile_wrapper)
+            env.define("CXX", compile_wrapper)
+            env.define("CPP", compile_wrapper)
+        tc.generate(env)
+
+        pkgconf = PkgConfigDeps(self)
+        pkgconf.generate()
+
+        deps = AutotoolsDeps(self)
+        deps.generate()
```

Comment:
```suggestion
```

there is only one dependency and it seems to be discovered by pkgconfig, so AutootolsDeps is useless, otherwise there is something wrong.